src/v1.js

```diff
@@ -136,19 +138,27 @@ api.declare({
   ].join('\n'),
 }, async function(req, res) {
   let prefix = req.query.prefix;
+  let continuationToken  = req.query.continuationToken || undefined;
+  let limit         = parseInt(req.query.limit || 10, 10);
+  let Client = this.Client;
+  let resolver = this.resolver;
 
-  // Load all clients
-  // TODO: as we acquire more clients, perform the prefix filtering in Azure
-  let clients = [];
-  await this.Client.scan({}, {
-    handler: client => {
+  let response = {clients: [], continuationToken};
+
+  while (true) {
+    let data = await Client.scan({}, {limit, continuation: response.continuationToken});
+    data.entries.forEach(client => {
       if (!prefix || client.clientId.startsWith(prefix)) {
-        clients.push(client.json(this.resolver));
+        response.clients.push(client.json(resolver));
       }
-    },
-  });
-
-  res.reply(clients);
+    });
```

Comment:
And here, before checking whether it's empty, set `continuationToken` to `data.continuationToken`.  Then if we loop again, we'll use the new continuationToken.

If `continuationToken` is not set at this point, we need to break from the loop and return an empty list to the user.

Reply:
Implemented this refactor.